### PR TITLE
Data Star

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
Fixes Plog and Data Star leaving behind muzzel traces of some projectiles

Adds a new boss mob for events Data Star

### CODE REVIEW

File: code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+	if(kcorp_moduals == 0 || marked_for_death)
+		..()
+		return
+	if(built_up_rage <= 1)
+		to_chat(world, "<b><font color='#ffaa00'>Data Collected, Qdel counter measure deployed.</font></b>")
+		built_up_rage += 3
+	if(built_up_rage > 2)
+		to_chat(world, "<b><font color='#ffaa00'>Qdel shield successful, Data Collected.</font></b>")
+
+/mob/living/carbon/superior_animal/robot/greyson/true_boss_data_star/forceMove()
```
Comment (Trilbyspaceclone) [inline code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm:375]:
```suggestion
/mob/living/carbon/superior_animal/robot/greyson/true_boss_data_star/forceMove(atom/destination, var/special_event, glide_size_override=0)
```

File: code/modules/admin/verbs/modifyvariables.dm [hunk 2]
```diff
@@ -1,6 +1,7 @@
 /client/var/global/list/forbidden_varedit_object_types = list(
 										/datum/admins,						//Admins editing their own admin-power object? Yup, sounds like a good idea.,
 										/obj/machinery/blackbox_recorder,	//Prevents people messing with feedback gathering
+										/mob/living/carbon/superior_animal/robot/greyson/true_boss_data_star, //Higher powers cant be cheated like this
```
Comment (Dimasw99) [inline code/modules/admin/verbs/modifyvariables.dm:4]: Event managers are going to hate you for this lmao

File: code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+This boss has 10 modes, a self revive system and a self healing
+As well as an "RCD" (Rapid Cover Devide)
+Will gain armor based on damage type
+Will gain EMP/Bomb protection
+This monster is borderline unkillable and will make players upset
+
+*/
+/mob/living/carbon/superior_animal/robot/greyson/true_boss_data_star
+	name = "Data Star"
+	desc = "A clearly well build drone by Greyson's, it sports a hologram eye that scans everything around. \
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm:23]:
```suggestion
	desc = "A clearly well built drone by Greyson, it sports a hologram eye that scans everything around. \
```

File: code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+	The unending feeling of dread builds up around this with its pure calutating killing gaze."
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm:24]:
```suggestion
	The unending feeling of dread builds up around this with its pure calculating killing gaze."
```

File: code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm [hunk 5]
```diff
@@ -0,0 +1,400 @@
...
+	destroy_surroundings = FALSE
+	armor = list(melee = -30, bullet = -30, energy = -30, bomb = -30, bio = 100, rad = 100) //Starting wise we are less then armorless
+
+	contaminant_immunity = TRUE
+
+	light_range = 6
+	light_color = COLOR_LIGHTING_BLUE_BRIGHT
+	mob_classification = CLASSIFICATION_SYNTHETIC
+
+	reload_message = "ejects a magazine as it loudly reloads with mechanical speed!"
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm:66]:
```suggestion
	reload_message = "ejects a spent box of ammo as it rearms its weaponry with machinelike speed!"
```

File: code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm [hunk 6]
```diff
@@ -0,0 +1,400 @@
...
+		return
+
+/mob/living/carbon/superior_animal/robot/greyson/true_boss_data_star/updatehealth()
+	set_sight(sight|SEE_TURFS|SEE_MOBS|SEE_OBJS) //wall hacks
+	health = maxHealth - getFireLoss() - getBruteLoss() - halloss //We cant have o2/clone/toxin damage affecet us at all
+	activate_ai()
+	process_med_hud(src,0)
+	if (health <= death_threshold && stat != DEAD)
+		if(kcorp_moduals > 0)
+			to_chat(world, "<b><font color='#ffaa00'>Data Star Morality Wounded, Data Collected.</font></b>")
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm:232]:
```suggestion
			to_chat(world, "<b><font color='#ffaa00'>Data Star Mortally Wounded, Data Collected.</font></b>")
```

Comment (cdb-is-not-good) [reply code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm:232]: either this, or "Data Star Critically Damaged, Data Collected."

File: code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm [hunk 7]
```diff
@@ -0,0 +1,400 @@
...
+			data_count += 4000 //We REALLY disliked that
+			kcorp_moduals -= 1
+		else
+			death()
+	handled_mode()
+	try_n_build()
+
+/mob/living/carbon/superior_animal/robot/greyson/true_boss_data_star/adjustBruteLoss(amount)
+	if(amount >= 1000)
+		to_chat(world,"<b><font color='#ffaa00'>Anti-Cheat Shielding Successfully Deplyed. Data Collected.</font></b>")
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm:244]:
```suggestion
		to_chat(world,"<b><font color='#ffaa00'>Anti-Cheat Shielding Successfully Deployed. Data Collected.</font></b>")
```

File: code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm [hunk 8]
```diff
@@ -0,0 +1,400 @@
...
+		built_up_rage += 1
+		if(built_up_rage > 2)
+			to_chat(world,"<b><font color='#ffaa00'>Lesser Being Orbitcal Beacon Shield Successfully Deplyed. Data Collected.</font></b>")
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm:247]:
```suggestion
			to_chat(world,"<b><font color='#ffaa00'>Lesser Being Orbital Beacon Shield Successfully Deployed. Data Collected.</font></b>")
```

File: code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm [hunk 9]
```diff
@@ -0,0 +1,400 @@
...
+			message_admins("<b><font color='#ffaa00'>Adminstation Being Data Collected.</font></b>")
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm:248]:
```suggestion
			message_admins("<b><font color='#ffaa00'>Administration Being Data Collected.</font></b>")
```

File: code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm [hunk 10]
```diff
@@ -0,0 +1,400 @@
...
+		return
+
+	..()
+	data_count += (amount*0.5)
+
+/mob/living/carbon/superior_animal/robot/greyson/true_boss_data_star/adjustFireLoss(amount)
+	if(amount >= 1000)
+		to_chat(world,"<b><font color='#ffaa00'>Anti-Cheat Shielding Successfully Deplyed. Data Collected.</font></b>")
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm:256]:
```suggestion
		to_chat(world,"<b><font color='#ffaa00'>Anti-Cheat Shielding Successfully Deployed. Data Collected.</font></b>")
```

File: code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm [hunk 11]
```diff
@@ -0,0 +1,400 @@
...
+		built_up_rage += 1
+		if(built_up_rage > 2)
+			to_chat(world,"<b><font color='#ffaa00'>Lesser Being Orbitcal Beacon Shield Successfully Deplyed. Data Collected.</font></b>")
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm:259]:
```suggestion
			to_chat(world,"<b><font color='#ffaa00'>Lesser Being Orbital Beacon Shield Successfully Deployed. Data Collected.</font></b>")
```

File: code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm [hunk 12]
```diff
@@ -0,0 +1,400 @@
...
+			message_admins("<b><font color='#ffaa00'>Adminstation Being Data Collected.</font></b>")
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm:260]:
```suggestion
			message_admins("<b><font color='#ffaa00'>Administration Being Data Collected.</font></b>")
```

File: code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm [hunk 13]
```diff
@@ -0,0 +1,400 @@
...
+			turrets_to_make -= 1
+			new /obj/random/turret/os(src.loc)
+
+/mob/living/carbon/superior_animal/robot/greyson/true_boss_data_star/handle_stunned()
+	if(data_count >= 1200 && stunned)
+		SetParalysis(0)
+		SetStunned(0)
+		SetWeakened(0)
+		SetDrowsyness(0)
+		to_chat(src,"<b><font color='#ffaa00'>Data Star Anti-Stun Deployed Successfully. Data Collected.</font></b>")
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm:318]: I might suggest rewording this and a few others to less directly refer to game mechanics. Maybe something like 'Data Star Disorientation Countermeasures Successful. Data Collected"

File: code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm [hunk 14]
```diff
@@ -0,0 +1,400 @@
...
+		data_count += 50 //We REALLY dislike being stunned
+		..()
+
+/mob/living/carbon/superior_animal/robot/greyson/true_boss_data_star/handle_weakened()
+	if(data_count >= 1200 && weakened)
+		SetParalysis(0)
+		SetStunned(0)
+		SetWeakened(0)
+		SetDrowsyness(0)
+		to_chat(src,"<b><font color='#ffaa00'>Data Star Anti-Weaken Deployed Successfully. Data Collected.</font></b>")
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm:330]: Ditto this, don't just refer to code stuff in  character it's a bit...ehh. 

"Data Star Anti-Mobility Countermeasures Deployed Successfully. Data Collected."

File: code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm [hunk 15]
```diff
@@ -0,0 +1,400 @@
...
+		dust(src)
+		..()
+
+//I do hope you knew what you were getting into by spawning this fine fellow
+/mob/living/carbon/superior_animal/robot/greyson/true_boss_data_star/Destroy()
+	if(kcorp_moduals == 0 || marked_for_death)
+		..()
+		return
+	if(built_up_rage <= 1)
+		to_chat(world, "<b><font color='#ffaa00'>Data Collected, Qdel counter measure deployed.</font></b>")
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm:370]:
```suggestion
		to_chat(world, "<b><font color='#ffaa00'>Data Collected, Qdel counter measure deployed.</font></b>")
```
Ehh....

**Review by cdb-is-not-good — COMMENTED** [review]
Still some weird grammar stuff but this should cover all the obvious typos/weird stuff. I want to strongly argue against using literal code terminology in game wherever possible(Qdel, Stun, Weaken, etc.)

File: code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm (hunk 14 above)
Comment (Trilbyspaceclone) [inline code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm:330]:
```suggestion
		to_chat(src,"<b><font color='#ffaa00'>Data Star Anti-Mobility Countermeasures Deployed Successfully. Data Collected.</font></b>")
```

File: code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm (hunk 13 above)
Comment (Trilbyspaceclone) [inline code/modules/mob/living/carbon/superior_animal/robots/subtype/greyson/types/data_star.dm:318]:
```suggestion
		to_chat(src,"<b><font color='#ffaa00'>Data Star Anti-Mobility Countermeasures Deployed Successfully. Data Collected.</font></b>")
```
